test/framework/seedregistrationframework.go

```diff
@@ -0,0 +1,383 @@
+	if f.GardenClient == nil {
+		return errors.New("no gardener client is defined")
+	}
+	if err := f.GardenClient.DirectClient().Get(ctx, client.ObjectKey{Name: seed.Name}, seed); err != nil {
+		return err
+	}
+	f.Seed = seed
+	return nil
+}
+
+//RegisterSeed registers a new seed and its secret reference (seed.Spec.SecretRef)
+//according to the configuration provided for the SeedRegistrationFramework
+func (f *SeedRegistrationFramework) RegisterSeed(ctx context.Context) (err error) {
+	if f.GardenClient == nil {
+		return errors.New("no gardener client is defined")
+	}
+
+	refShoot := &gardencorev1beta1.Shoot{}
+	if err = f.GardenClient.DirectClient().Get(ctx, kutil.Key(f.Config.ShootedSeedNamespace, f.Config.ShootedSeedName), refShoot); err != nil {
+		return err
+	}
+
+	if deploy, _ := strconv.ParseBool(f.Config.DeployGardenletOnShoot); deploy {
+		return f.deployGardenletOnShoot(ctx, refShoot)
+	}
+
+	secretBinding := &gardencorev1beta1.SecretBinding{}
+	if err = f.GardenClient.DirectClient().Get(ctx, kutil.Key(f.Config.ShootedSeedNamespace, f.Config.SecretBinding), secretBinding); err != nil {
+		return err
+	}
+
+	f.Seed = f.buildSeedObject(*secretBinding, *refShoot)
+
+	if err = f.generateSeedSecret(ctx, secretBinding); err != nil {
+		f.Logger.Fatalf("Cannot generate seed secret %s", err)
+		return err
+	}
+	f.Logger.Infof("Creating seed secret %s in namespace %s", f.Secret.GetName(), f.Secret.GetNamespace())
+	//Apply secret to the cluster
+	if err = client.IgnoreNotFound(f.GardenClient.DirectClient().Create(ctx, f.Secret)); err != nil {
+		f.Logger.Fatalf("Cannot create seed secret %s: %s", f.Secret.GetName(), err)
+		return err
+	}
+	f.Logger.Infof("Secret resource %s was created!", f.Secret.Name)
+
+	f.Logger.Infof("Registering seed %s", f.Seed.GetName())
+	if err := PrettyPrintObject(f.Seed); err != nil {
+		f.Logger.Fatalf("Cannot decode seed %s: %s", f.Seed.GetName(), err)
+		return err
+	}
+	return f.GardenerFramework.CreateSeed(ctx, f.Seed)
+}
+
+func (f *SeedRegistrationFramework) deployGardenletOnShoot(ctx context.Context, refShoot *gardencorev1beta1.Shoot) (err error) {
+	annotations := make(map[string]string)
+	annotations["shoot.gardener.cloud/use-as-seed"] = "true,protected,with-secret-ref"
+	f.Logger.Infof("Annotating shoot with: %v", annotations)
+	if err = f.GardenerFramework.AnnotateShoot(ctx, refShoot, annotations); err != nil {
+		f.Logger.Errorf("Unable to annotate shoot due to: %s", err.Error())
+		return err
+	}
+	return f.GardenerFramework.UpdateShoot(ctx, refShoot, func(shoot *gardencorev1beta1.Shoot) error {
+		vpa := &gardencorev1beta1.VerticalPodAutoscaler{
+			Enabled: true,
+		}
+		shoot.Spec.Kubernetes.VerticalPodAutoscaler = vpa
+
+		return nil
+	})
+}
+
+func (f *SeedRegistrationFramework) buildSeedObject(secretBinding gardencorev1beta1.SecretBinding, refShoot gardencorev1beta1.Shoot) *gardencorev1beta1.Seed {
+	var (
+		seed    = &gardencorev1beta1.Seed{}
```

Comment:
Wouldn't be more readable to initialize the Seed inline? For example
```suggestion
		seed    = &gardencorev1beta1.Seed{
			ObjectMeta: metav1.ObjectMeta{
				Name: "foo",
			},
			Spec: gardencorev1beta1.SeedSpec{
				// ...
			},
		}
```